### PR TITLE
Map Power BI report pages to charts

### PR DESCRIPTION
**Pull request by mars-lan**
### 🤔 Why?

Power BI's [REST API](https://docs.microsoft.com/en-us/rest/api/power-bi/) doesn't include any visualization information (charts) for [reports](https://docs.microsoft.com/en-us/power-bi/fundamentals/service-basic-concepts#reports). However, the name of the pages (tabs) in the report is considered a good proxy for the visualizations and should therefore be mapped to charts.

### 🤓 What?

As titled. Use the [Get Pages in Group](https://docs.microsoft.com/en-us/rest/api/power-bi/reports/get-pages-in-group) API to retrieve the information, which requires additional permissions. 

### 🧪 Tested?

Tested against a production deployment.

### CODE REVIEW

File: metaphor/power_bi/extractor.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import json
 import tempfile
+from multiprocessing import AuthenticationError
```
Comment (elic-eon) [inline metaphor/power_bi/extractor.py:3]: Does this error class show the useful error message?

File: metaphor/power_bi/extractor.py [hunk 2]
```diff
@@ -152,6 +159,21 @@ class WorkspaceInfo(BaseModel):
     dashboards: List[WorkspaceInfoDashboard] = []
 
 
+class NotAuthorizedError(Exception):
```
Comment (elic-eon) [inline metaphor/power_bi/extractor.py:162]: Do we use this class?

File: metaphor/power_bi/README.md [hunk 3]
```diff
@@ -21,6 +22,8 @@ For example,
 
 ![](https://docs.microsoft.com/en-us/power-bi/enterprise/media/read-only-apis-service-principal-auth/allow-service-principals-tenant-setting.png)
 
+4. [Add the service principal](https://docs.microsoft.com/en-us/power-bi/developer/embedded/embed-service-principal#step-4---add-the-service-principal-to-your-workspace) to all workspaces of interest, except "My Workspaces". Note that it may take an hour before the permission is fully propagated.
```
Comment (elic-eon) [inline metaphor/power_bi/README.md:25]: I think there is no other way to get rid of adding the service principal to all workspaces.

**Review by elic-eon — COMMENTED** [review]
Otherwise, LGTM

File: metaphor/power_bi/extractor.py (hunk 2 above)
Comment (mars-lan) [reply metaphor/power_bi/extractor.py]: Wrong class :(

File: metaphor/power_bi/README.md (hunk 3 above)
Comment (mars-lan) [reply metaphor/power_bi/README.md:25]: The read-only Admin APIs don't require this. Unfortunately, they don't return the page information, so we had to fallback to the non-admin Power BI APIs.

**Pull request merged**